
Compile-fix for commit 02aef1f4fa1b6a46a0e183bc021719ff92cc4974
(Btrfs: Fix the problem that the dirty flag of dev stats is cleared):

kernel 3.15 renamed various smp_mb__* definitions and we need to revert
to their previous versions.

-HH

--- linux-3.14.17-broken/fs/btrfs/volumes.h	2014-08-27 02:01:37.307304523 +0200
+++ linux-3.14.17-ok/fs/btrfs/volumes.h	2014-08-27 01:56:17.731377155 +0200
@@ -366,7 +366,7 @@
 				      int index)
 {
 	atomic_inc(dev->dev_stat_values + index);
-	smp_mb__before_atomic();
+	smp_mb__before_atomic_inc();
 	atomic_inc(&dev->dev_stats_ccnt);
 }
 
@@ -382,7 +382,7 @@
 	int ret;
 
 	ret = atomic_xchg(dev->dev_stat_values + index, 0);
-	smp_mb__before_atomic();
+	smp_mb__before_atomic_inc();
 	atomic_inc(&dev->dev_stats_ccnt);
 	return ret;
 }
@@ -391,7 +391,7 @@
 				      int index, unsigned long val)
 {
 	atomic_set(dev->dev_stat_values + index, val);
-	smp_mb__before_atomic();
+	smp_mb__before_atomic_inc();
 	atomic_inc(&dev->dev_stats_ccnt);
 }
